File: app/src/main/java/com/tangem/di/NavigatorComponent.java
package com.tangem.di;

import com.tangem.presentation.activity.EmptyWalletActivity;
import com.tangem.presentation.activity.LoadedWalletActivity;
import com.tangem.presentation.activity.LogoActivity;
import com.tangem.presentation.activity.MainActivity;
import com.tangem.presentation.activity.PrepareCryptonitOtherApiWithdrawalActivity;
import com.tangem.presentation.activity.PrepareKrakenWithdrawalActivity;
import com.tangem.presentation.activity.PrepareTransactionActivity;
import com.tangem.presentation.activity.PurgeActivity;
import com.tangem.presentation.activity.VerifyCardActivity;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {NavigatorModule.class})
public interface NavigatorComponent {

    void inject(LogoActivity activity);

    void inject(MainActivity activity);

    void inject(PurgeActivity activity);

    void inject(PrepareTransactionActivity activity);

    void inject(PrepareCryptonitOtherApiWithdrawalActivity activity);

    void inject(PrepareKrakenWithdrawalActivity activity);

    void inject(LoadedWalletActivity activity);

    void inject(VerifyCardActivity activity);

    void inject(EmptyWalletActivity activity);

}

File: app/src/main/java/com/tangem/di/NavigatorComponent.kt
package com.tangem.di

import com.tangem.ui.activity.EmptyWalletActivity
import com.tangem.ui.activity.LoadedWalletActivity
import com.tangem.ui.activity.LogoActivity
import com.tangem.ui.activity.MainActivity
import com.tangem.ui.activity.PrepareCryptonitOtherApiWithdrawalActivity
import com.tangem.ui.activity.PrepareKrakenWithdrawalActivity
import com.tangem.ui.PrepareTransactionActivity
import com.tangem.ui.activity.PurgeActivity
import com.tangem.ui.activity.VerifyCardActivity

import javax.inject.Singleton

import dagger.Component

@Singleton
@Component(modules = [NavigatorModule::class])
interface NavigatorComponent {

    fun inject(activity: LogoActivity)

    fun inject(activity: MainActivity)

    fun inject(activity: PurgeActivity)

    fun inject(activity: PrepareTransactionActivity)

    fun inject(activity: PrepareCryptonitOtherApiWithdrawalActivity)

    fun inject(activity: PrepareKrakenWithdrawalActivity)

    fun inject(activity: LoadedWalletActivity)

    fun inject(activity: VerifyCardActivity)

    fun inject(activity: EmptyWalletActivity)

}